### DISCUSSION TITLE
How to delete user from React or Nextjs app?

### DISCUSSION THREAD

**Discussion by alexalexyang** [zitadel/zitadel#6771] [Q&A] [answered]
I can't figure out how to delete an authenticated user from a React/Nextjs front end. At first I thought `userManager.removeUser()` on the front end would delete the user but it didn't do anything.

The [API docs](https://zitadel.com/docs/apis/resources/auth/auth-service-remove-my-user) provides this example:

```node
const axios = require('axios');

let config = {
  method: 'get',
maxBodyLength: Infinity,
  url: 'https://$CUSTOM-DOMAIN/auth/v1/users/me',
  headers: { 
    'Accept': 'application/json', 
    'Authorization': 'Bearer <TOKEN>'
  }
};

axios(config).then(...)
```

I suppose this is for the back end so it'd work in a Next.js API route. But how do we pass in any user identification? Do we get `sub` or another property from `userManager.getUser().profile` and replace `me` in the URL with it? That gets a lot of `Not Found` errors so I must be wrong.

If we delete the user from the console (`https://.../ui/users/console/me?id=...), we get this request:

```javascript
await fetch("https://.../zitadel.management.v1.ManagementService/RemoveUser", {
        "credentials": "include",
        "headers": {
            ...
            "Authorization": "Bearer ...",
            ...
        },
        "body": "\u0000\u0000\u0000\u0000\u0014\n\u00122...",
        "method": "POST",
        "mode": "cors"
    });
```

Should we use `https://.../zitadel.management.v1.ManagementService/RemoveUser` to delete users?

How do we do it?

Thanks for your patience 🙏🏼.

**Comment by hifabienne** [comment]
Hei the auth api is for authenticated users, if they want to do on their own user.
You can use the following request: https://zitadel.com/docs/apis/resources/auth/auth-service-remove-my-user
This endpoint requires a permission, so make sure the user has the role: SELF_MANAGEMENT_GLOBAL

**Reply by alexalexyang** [reply, marked as answer]
Hi. I just read [this](https://zitadel.com/docs/guides/manage/console/managers) and realise maybe the [auth-service-remove-my-user endpoint](https://zitadel.com/docs/apis/resources/auth/auth-service-remove-my-user) isn't what I need. It seems SELF_MANAGEMENT_GLOBAL and other such permissions would mean "users are allowed to login into your instance console and edit certain parts of the configuration". I don't want ordinary users to be managers who are able to log into my instance console. I'd like to simply delete them when they press a delete button on a front end.

Does this mean I need to use a Service User with role Org User Manager to delete ordinary human users? However, I've scrolled through the API docs [here](https://zitadel.com/docs/apis/introduction) and don't see that that's possible.

Am I misunderstanding something?

Thanks.

Update: I think it's [this endpoint](https://zitadel.com/docs/apis/resources/mgmt/management-service-remove-user) I'm looking for. Missed this on first research. I suppose I can use a Service User to use this endpoint to delete human users. I will try this probably over the weekend.

**Reply by hifabienne** [reply]
Hei yes it depends a little on your use case.
Users are always able to login to the management console. Depending on their roles they will see more or less.

As not in every use case you want the users beeing able to delete them self, we had to implement a permission for that.
If you do not want them to give that permission, your only possibility is to do it with the endpoint you mentioned and with a service account that has the rights to do that.
